File: SortingAlgorithms/python/QuickSort.py
def quicksort(blist, start, end):
    ##'Sorts the list from indexes start to end - 1 inclusive.'
    if end - start > 1:
        p = partition(blist, start, end)
        quicksort(blist, start, p)
        quicksort(blist, p + 1, end)
 
 
def partition(blist, start, end):
    pivot = blist[start]
    i = start + 1
    j = end - 1
 
    while True:
        while (i <= j and blist[i] <= pivot):
            i = i + 1
        while (i <= j and blist[j] >= pivot):
            j = j - 1
 
        if i <= j:
            blist[i], blist[j] = blist[j], blist[i]
        else:
            blist[start], blist[j] = blist[j], blist[start]
            return j
 
 
blist = input('Enter the list of numbers: ').split()
blist = [int(x) for x in blist]
quicksort(blist, 0, len(blist))
print('Sorted list: ', end='')
print(blist)

## The user has to enter a list of numbers.
## The list is passed to the quicksort function.
## The sorted list is displayed.

File: SortingAlgorithms/C++/quickSort.cpp
#include<iostream>
#include<functional>
using namespace std;

int Partition(int *a,int start,int end)
{
	int pivot=a[end];
	int pIndex=start;

	for(int i=start;i<end;i++)
	{
		if(a[i]<pivot)
		{
			if(pIndex!=i)
				swap(a[i],a[pIndex]);
			pIndex=pIndex+1;
		}
	}

	swap(a[pIndex],a[end]);
	return pIndex;

	
}

void Quicksort(int *a,int start,int end)
{

	if(start<end)
	{
		int pIndex=Partition(a,start,end);
		Quicksort(a,start,pIndex-1);
		Quicksort(a,pIndex+1,end);
	}
 
}

int main()
{
	int a[]={7,3,9,1,5,2};
	int n=sizeof(a)/sizeof(a[0]);

	Quicksort(a,0,n-1);

	for(int i=0;i<n;i++)
		cout << a[i] <<" ";

	return 0;
}